RxTest/Recorded.swift

```diff
@@ -32,6 +32,40 @@ extension Recorded {
     }
 }
 
+extension Recorded {
+    
+    /**
+     Factory method for an `.next` event recorded at a given time with a given value.
+     
+     - parameter time: Recorded virtual time the `.next` event occurs.
+     - parameter element: Next sequence element.
+     - returns: Recorded event in time.
+     */
+    public static func next<T>(_ time: TestTime, _ element: T) -> Recorded<Event<T>> where Value == Event<T> {
```

Comment:
Wouldn't it be more consistent with api guidelines if we used:
`public static func next<T>(time: TestTime, element: T) -> Recorded<Event<T>> where Value == Event<T>`
`public static func completed<T>(time: TestTime, type: T.Type = T.self) -> Recorded<Event<T>> where Value == Event<T>`
`public static func error<T>(time: TestTime, error: Swift.Error, type: T.Type = T.self) -> Recorded<Event<T>> where Value == Event<T> `

?

We can use regex to find replace all of the places.

Reply:
Yes, I totally agree. 👍

I've added the parameter label as suggested:

```swift
// This is call side

// Before:

let recordedEvents: [Recorded<Event<Int>>] = [
    next(205, 1),
    next(210, 2),
    next(240, 3),
    error(500, testError),
    completed(600)
]

// After:

let recordedEvents: [Recorded<Event<Int>>] = [
    .next(time: 205, element: 1),
    .next(time: 210, element: 2),
    .next(time: 240, element: 3),
    .error(time: 500, error: testError),
    .completed(time: 600)
]

``` 

Then i have a new idea if we can remove the `element`, `error` parameter label:

```swift

// Before:

let recordedEvents: [Recorded<Event<Int>>] = [
    .next(time: 205, element: 1),
    .next(time: 210, element: 2),
    .next(time: 240, element: 3),
    .error(time: 500, error: testError),
    .completed(time: 600)
]

// After:

let recordedEvents: [Recorded<Event<Int>>] = [
    .next(time: 205, 1),
    .next(time: 210, 2),
    .next(time: 240, 3),
    .error(time: 500, testError),
    .completed(time: 600)
]

```

This may add readability for test purpose. And will align the api when create `Event`:

```swift
let events: [Event<Int>] = [
    .next(1),
    .next(2),
    .next(3),
    .error(testError),
    .completed
]

let recordedEvents: [Recorded<Event<Int>>] = [
    .next(time: 205, 1),
    .next(time: 210, 2),
    .next(time: 240, 3),
    .error(time: 500, testError),
    .completed(time: 600)
]
```

Is this idea reasonable? 🤔

Reply:
Just my $.02: adding a named `time` parameter will just make it more tedious to write the hundreds of such lines that need to be written for tests. Writing tests is long enough, I don't think we need this useless burden added.

Reply:
Hi @fpillet Thanks for your valuable suggestions.

Now, There are three choices here:

1. function  with all named parameters:

    ```swift
    let recordedEvents: [Recorded<Event<Int>>] = [
        .next(time: 205, element: 1),
        .next(time: 210, element: 2),
        .next(time: 240, element: 3),
        .error(time: 500, error: testError),
        .completed(time: 600)
    ]
    ````

2. function  with just named `time` parameter:

    ```swift
    let recordedEvents: [Recorded<Event<Int>>] = [
        .next(time: 205, 1),
        .next(time: 210, 2),
        .next(time: 240, 3),
        .error(time: 500, testError),
        .completed(time: 600)
    ]
    ````

3. function  without any named parameter:

    ```swift
    let recordedEvents: [Recorded<Event<Int>>] = [
        .next(205, 1),
        .next(210, 2),
        .next(240, 3),
        .error(500, testError),
        .completed(600)
    ]
    ````

And option 3 seems to be light burden when writing test code.
@kzaher what do you think of it? 🦄

Reply:
I know the conventions mandates naming parameters, but the sheer volume of such calls that need to be written for tests make it somewhat impractical to follow convention in this case... thanks!

Reply:
I'm fine with option 3 since it's practical, although I'm not sure it's according to swift guidelines.

Reply:
I think specifically for testing things option 3 makes more sense. 

The other options constitute so much boilerplate in this case where you have to repeatedly make use of this method. 

I think Guidelines, like any others, can be steered away from if there is a clearly good reason to, like this one :) just my 2 cents